## Ejercicios realizados sobre el Análisis Exploratorio Realizado


In [1]:
import numpy as np
import pandas as pd
import matplotlib
import matplotlib.pyplot as plt
import scipy.stats as stats
import statsmodels.stats.diagnostic as diag
import statsmodels.api as sm
import seaborn as sns

**a. ¿Cuáles son las 10 películas que contaron con más presupuesto?**


In [8]:
df = pd.read_csv('movies.csv', encoding='ISO-8859-1')

# Sort the DataFrame by the 'budget' column in descending order
sorted_df = df.sort_values(by='budget', ascending=False)

# Select the top 10 movies with the highest budgets
top_10_budget_films = sorted_df[['originalTitle', 'budget']].head(10)

# Format the 'budget' values as currency (e.g., $1,000,000)
top_10_budget_films['budget'] = top_10_budget_films['budget'].apply(lambda x: f"${x:,.0f}")

# Display the resulting DataFrame with the 'originalTitle' and formatted 'budget' columns
display(top_10_budget_films)

originalTitle        budget
716   Pirates of the Caribbean: On Stranger Tides  $380,000,000
4710                      Avengers: Age of Ultron  $365,000,000
5952                            Avengers: Endgame  $356,000,000
5953                       Avengers: Infinity War  $300,000,000
163      Pirates of the Caribbean: At World's End  $300,000,000
4953                               Justice League  $300,000,000
607                              Superman Returns  $270,000,000
3791                                      Tangled  $260,000,000
7134                                The Lion King  $260,000,000
280                                  Spider-Man 3  $258,000,000

**Análisis de las Películas con Mayor Presupuesto**

<small> 
La tabla anterior presenta las 10 películas con los mayores presupuestos en nuestro dataset. "Pirates of the Caribbean: On Stranger Tides" lidera la lista con un presupuesto de $380 millones, seguido de cerca por "Avengers: Age of Ultron" con $365 millones y "Avengers: Endgame" con $356 millones. Estos elevados presupuestos reflejan el alto coste asociado con la producción de películas de gran escala, que a menudo incluyen extensos efectos especiales, grandes elencos y equipos técnicos, así como locaciones globales.

Este grupo de películas destaca por ser predominantemente del género de acción y aventura, lo cual es indicativo de la tendencia de la industria cinematográfica a invertir considerablemente en películas que apuntan a generar grandes taquillas a nivel mundial. La presencia repetida de franquicias como "Avengers" y "Pirates of the Caribbean" subraya la apuesta de los estudios por secuelas y series de películas que ya han demostrado ser exitosas comercialmente.
</small>


**b. ¿Cuáles son las 10 películas que más ingresos tuvieron?**


In [ ]:
# Read the CSV file into a DataFrame
df = pd.read_csv('movies.csv', encoding='ISO-8859-1')

# Convert the 'revenue' column to numeric, coercing errors to NaN
df['revenue'] = pd.to_numeric(df['revenue'], errors='coerce')

# Sort the DataFrame by the 'revenue' column in descending order and select the top 10 movies
top_revenue_films = df.sort_values(by='revenue', ascending=False)[['originalTitle', 'revenue']].head(10)

# Format the 'revenue' values as currency (e.g., $1,000,000)
top_revenue_films['revenue'] = top_revenue_films['revenue'].apply(lambda x: f"${x:,.0f}")

# Display the resulting DataFrame with the 'originalTitle' and formatted 'revenue' columns
display(top_revenue_films)

originalTitle         revenue
3210                        Avatar  $2,847,246,203
5952             Avengers: Endgame  $2,797,800,564
307                        Titanic  $2,187,463,944
4947  Star Wars: The Force Awakens  $2,068,223,624
5953        Avengers: Infinity War  $2,046,239,637
4914                Jurassic World  $1,671,713,208
7134                 The Lion King  $1,667,635,327
9049       Spider-Man: No Way Home  $1,631,853,496
3397                  The Avengers  $1,518,815,515
5087                     Furious 7  $1,515,047,671

**Análisis de las Películas con Mayores Ingresos**

<small>
La tabla anterior muestra las 10 películas que han generado los mayores ingresos en nuestro dataset. "Avatar" encabeza la lista con ingresos de $2,847 millones, seguido por "Avengers: Endgame" con $2,798 millones y "Titanic" con $2,187 millones. Estas cifras destacan no solo el éxito comercial de estas películas, sino también su amplio alcance y popularidad global.

Este conjunto de películas es principalmente de los géneros de acción, ciencia ficción y drama, lo que refleja la preferencia de la audiencia por historias épicas y efectos visuales espectaculares. La presencia de múltiples películas de la saga "Avengers" y otros títulos icónicos como "Star Wars: The Force Awakens" y "Jurassic World" demuestra la tendencia de los estudios de cine a invertir en franquicias de gran escala que prometen no solo retorno financiero, sino también un fuerte seguimiento de fanáticos a nivel mundial.
</small>


**c. ¿Cuál es la película que más votos tuvo?**


In [13]:
df = pd.read_csv('movies.csv', encoding='ISO-8859-1')

# Convert the 'voteCount' column to numeric, coercing errors to NaN
df['voteCount'] = pd.to_numeric(df['voteCount'], errors='coerce')

# Sort the DataFrame by the 'voteCount' column to find the top voted film
top_voted_film = df.loc[df['voteCount'].idxmax()]

print("La película con más votos es:")
display(df.loc[[df['voteCount'].idxmax()]][['originalTitle', 'voteCount']])

La película con más votos es:


originalTitle  voteCount
3511     Inception      30788

**Película con Mayor Número de Votos**

<small> La película que ha recibido el mayor número de votos dentro del dataset es **Inception**, con un total de **30788** votos. Esto indica no solo una alta participación de la audiencia sino también un gran interés y compromiso del público con esta película, reflejando su popularidad y resonancia cultural. </small>
